Code/Tools/ProjectManager/Source/ProjectButtonWidget.h

```diff
@@ -106,17 +118,33 @@ namespace O3DE::ProjectManager
     private:
         void enterEvent(QEvent* event) override;
         void leaveEvent(QEvent* event) override;
-        void ShowWarning(bool show, const QString& warning);
-        void ShowDefaultBuildButton();
+
+        void ShowReadyState();
+        void ShowLaunchingState();
+        void ShowBuildRequiredState();
+        void ShowBuildingState();
+        void ShowBuildFailedState();
+        void ShowMessage(const QString& message = {}, const QString& submessage = {});
+        void ShowWarning(const QString& warning = {});
+        void ShowBuildButton();
+        void SetLaunchingEnabled(bool enabled);
+        void SetProjectBuilding(bool isBuilding);
+        void HideContextualLabelButtonWidgets();
 
         QMenu* CreateProjectMenu();
 
         ProjectInfo m_projectInfo;
+        QUrl m_logUrl;
```

Comment:
Are we duplicating a QUrl here? I noticed `m_projectInfo` also has a `m_logUrl` - can we just use it?

Reply:
Looking through the code it doesn't look like m_logUrl is ever used to successfully pass the log url to the ProjectButton, so we still probably need to setter. We could definitely use that to store the logUrl we receive thought!